submissions/DmytryjK/js-oop-exercise-tiny/index.js

```diff
@@ -0,0 +1,97 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+
+// ======== OUTPUT ========
+/* Use print(message) for output.
+   Default tag for message is <pre>. Use print(message,'div') to change containing element tag.
+
+   Message can contain HTML markup. You may also tweak index.html and/or styles.css.
+   However, please, REFRAIN from improving visuals at least until your code is reviewed
+   so code reviewers might focus on a single file that is index.js.
+   */
+
+/* Print examples:
+   print('ABC');
+   print('<strong>ABC</strong>');
+   print('<strong>ABC</strong>', 'div');
+
+   print('human; John; male; 2; 2; Hello world!; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny', 'div');
+   */
+
+class Inhabitant {
+   constructor(name, spacies, gender, saying, legs) {
+      this.name = name;
+      this.spacies = spacies;
+      this.gender = gender;
+      this.saying = saying;
+      this.legs = legs;
+      this.property = ['name', 'spacies', 'gender', 'saying', 'legs'];
+   }
+   get props() {
+      return this.property.map(key => this[key]).join('; ');
+   }
+}
+
+class Human extends Inhabitant {
+   constructor (name, gender, saying) {
+      super(name, 'human', gender, saying, '2');
+      this.hands = 2;
+   }
+   get props() {
+      return super.props + `; ${this.hands}`;
+   }
+}
+
+class Dog extends Inhabitant {
+   constructor (name, gender, saying) {
+      super(name, 'dog', gender, saying, '4');
+   }
+   get props() {
+      return super.props;
+   }
+}
+
+class Cat extends Inhabitant {
+   constructor (name, gender, saying) {
+      super(name, 'cat', gender, saying, '4');
+   }
+   get props() {
+      return super.props;
+   }
+}
+
+class CatWoman extends Cat {
+   constructor (name, gender) {
+      super(name, gender, cat.saying);
+      this.spacies = 'catWoman';
+      this.legs = 2;
+      this.hands = 2;
+   }
```

Comment:
Looks like CatWoman has more in common with humans than with cats.
Also note that we had to "re-add" `hands` and overwrite `legs`.
Actually OOP is not good with hybrids. The proper pattern for hybrids is composition.
You will have a chance to study composition patterns.
For now we done as much as possible in these regards with OOP.
Take this learning. Also check class static methods and properties. This is yet another way to "borrow" behaviour from another class.